copy/copy.go

```diff
@@ -597,15 +647,35 @@ func (c *copier) createProgressBar(pool *mpb.Progress, info types.BlobInfo, kind
 		prefix = prefix[:maxPrefixLen]
 	}
 
-	bar := pool.AddBar(info.Size,
-		mpb.BarClearOnComplete(),
+	options := []mpb.BarOption{
 		mpb.PrependDecorators(
 			decor.Name(prefix),
 		),
-		mpb.AppendDecorators(
-			decor.OnComplete(decor.CountersKibiByte("%.1f / %.1f"), " "+onComplete),
-		),
-	)
+	}
+	options = append(options, barOpts...)
+
+	// Fillers are used for placeholder bars that are later replaced by a
+	// common progress bar.  Hence, only append a counter decorator when
+	// filler == nil.
+	if filler == "" {
+		options = append(options,
+			mpb.AppendDecorators(
+				decor.OnComplete(decor.CountersKibiByte("%.1f / %.1f"), " done"),
+			))
+		options = append(options, mpb.BarClearOnComplete())
+	}
+
+	var bar *mpb.Bar
+	if filler != "" {
+		barFiller := mpb.FillerFunc(
```

Comment:
(BTW, if I understand the idea correctly, we don’t need to replace the bars only to change the message; the `FillerFunc` could just `fmt.Fprint(*someVariable)` and we could update the variable (+ locking in both the reader and writer).

That does not look like an improvement right now, but if the logic moved into `storageImageDestination.TryReuseBlob`, it could make the API to update the progress bar with status messages notably easier.)

Reply:
I think it's possible to do that but this would not work when replacing a "placeholder" with the real progress bar, which does not specify a filler. I prefer to keep it as is (now with a dedicated method to create a replacement bar).

Reply:
Sure, the real progress bar can be a different different animal.

The `fmt.Fprintf(*someVariable)` approach seems attractive in that it would give `GetBlob` (for status about mirrors) and `TryReusingBlob` (for all of this code, if we manage to move it into the transport) a pretty nice way to report progress: we could give them a
```go
type UserProgressNotifier interface {
   Update(string message)
}
```